### PR TITLE
tests: merge Linux wheezy workflow with main workflow

### PR DESCRIPTION
**Pull request by iMichka**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/binutils.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@ class Binutils < Formula
   url "https://ftp.gnu.org/gnu/binutils/binutils-2.36.1.tar.xz"
   mirror "https://ftpmirror.gnu.org/binutils/binutils-2.36.1.tar.xz"
   sha256 "e81d9edf373f193af428a0f256674aea62a9d74dfe93f65192d4eae030b0f3b0"
+  revision 1
   license all_of: ["GPL-2.0-or-later", "GPL-3.0-or-later", "LGPL-2.0-or-later", "LGPL-3.0-only"]
```
Comment (carlocab) [inline Formula/binutils.rb:8]:
```suggestion
  license all_of: ["GPL-2.0-or-later", "GPL-3.0-or-later", "LGPL-2.0-or-later", "LGPL-3.0-only"]
  revision 1
```

**Comment by iMichka** [comment]
I added the label to the label list, I guess it was not able to add the label automatically. Weird.

**Comment by dawidd6** [comment]
I think we gonna have to tweak the Dockerfile. First of all to include homebrew-core in it. Then move it to Homebrew/brew repo and name it something like Dockerfile.wheezy and build, upload it in the CI to ghcr.io.

**Comment by iMichka** [comment]
I see. The file is in https://github.com/Homebrew/homebrew-linux-dev/blob/master/Dockerfile. You want to move it to Homebrew/brew to rebuild it on a regular basis, so we always have a somewhat up to date homebrew-core repo in there?

I think building this docker image is slow, and I am worried that one day the servers for wheezy might be shut down, resulting in a broken Homebrew/brew build. This docker image is more a "time capsule" that does not need new rebuilds if not necessary.

**Comment by dawidd6** [comment]
I don't think the Debian archive will be shut down, neither will the maintainers delete `wheezy/` packages from there, cause as one can see there is still `potato/` release in the archive which is even older. But I understand your concern, there is no need to rebuild this image very often, maybe we can agree on building it on brew new release only?

For sure we need homebrew-core copy in this image, as I don't think there is one.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -169,6 +178,24 @@ jobs:
           # Fix working directory permissions
           docker exec --user root linuxbrew chmod 777 /tmp/bottles
 
+      - name: Run Linux Wheezy Docker container
+        if: runner.os == 'Linux' && needs.pre_tests.outputs.linux-wheezy == 'true'
+        env:
+          HOMEBREW_ON_DEBIAN7: 1
```
Comment (dawidd6) [inline .github/workflows/tests.yml:184]: Those 3 env vars will/should be set in Dockerfile

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -85,6 +86,14 @@ jobs:
               core.setOutput('force-linux', 'false')
             }
 
+            if (label_names.includes('CI-linux-wheezy')) {
+              console.log('CI-linux-wheezy label found. Running Linux wheezy builds.')
+              core.setOutput('linux-wheezy', 'true')
```
Comment (dawidd6) [inline .github/workflows/tests.yml:91]: We can output the image name here, and the just use the output in `docker run`.
Once those 3 env vars below are gone ofc.

**Comment by dawidd6** [comment]
We might need to rename the container to something unique like `${{github.sha}}`, cause multiple running containers on our self-hosted runner could be problematic.

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -169,6 +178,24 @@ jobs:
           # Fix working directory permissions
           docker exec --user root linuxbrew chmod 777 /tmp/bottles
 
+      - name: Run Linux Wheezy Docker container
```
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml:181]: This duplication here is a bit nasty. I don't mind it shipping as-is for now but would really love to get all this logic into Homebrew/actions.

**Review by MikeMcQuaid — DISMISSED** [review]
Fine as-is for me but some cleanup suggestions here. More in https://github.com/Homebrew/homebrew-core/pull/79894

**Comment by dawidd6** [comment]
Imma merge this and create a new PR with some fixes.

File: .github/workflows/tests.yml (hunk 4 above)
Comment (dawidd6) [reply .github/workflows/tests.yml]: Duplication is no more. But agree on the action side.

Comment (dawidd6) [reply .github/workflows/tests.yml]: Resolved for now, to not clutter the view.

**Pull request merged**